File: timus.ru/vol1(1000-1099)/1036.py
n, s = map(int, input().split())

if (s & 1):
    print(0)
    exit()

s //= 2
max_sum = 9 * n

if (s > max_sum):
    print(0)
    exit()

dp = [[0 for i in range(0, max_sum + 1)] for j in range(n)]

for i in range(10):
    dp[0][i] = 1

for i in range(1, n):
    for j in range((i + 1) * 9 + 1):
        for k in range(min(j, 9) + 1):
            dp[i][j] += dp[i - 1][j - k]

print(dp[n - 1][s] * dp[n - 1][s])

File: timus.ru/vol1(1000-1099)/1036.cpp
#include <bits\stdc++.h>

using namespace std;

using ui64 = uint64_t;

int main()
{
    int n, s;
    cin >> n >> s;
    if (s & 1)
    {
        cout << "0\n";
        exit(0);
    }
    s /= 2;
    int max_sum = 9 * n;
    if (s > max_sum)
    {
        cout << "0\n";
        exit(0);
    }
    vector<vector<ui64>> dp(n, vector<ui64>(max_sum + 1));
    for (int i = 0; i < 10; ++i)
    {
        dp[0][i] = 1;
    }
    for (int i = 1; i < n; ++i)
    {
        for (int j = 0; j <= (i + 1) * 9; ++j)
        {
            for (int k = 0; k <= min(j, 9); ++k)
            {
                dp[i][j] += dp[i - 1][j - k];
            }
        }
    }
    cout << dp[n - 1][s] * dp[n - 1][s] << '\n';
}